### PR TITLE
feat: add edit option in suggested queries in hint commands

### LINKED ISSUE #32075: [Task]: Show the edit option in the suggested queries in the hint dropdown
### SubTasks

image

[Notion link ](https://www.notion.so/appsmith/Add-Edit-CTA-on-one-click-binding-options-for-API-Query-0a6fa24dfedd471594c65be38579db8b)

### PR DESCRIPTION
**Pull request by AmanAgarwal041**
## Description
This PR  introduces a new way to edit the queries directly from slash hint commands that edit button appears on hover of the hint. This also introduces an event `EDIT_ACTION_CLICK` as emitted from different other places of our application  with all the similar properties except the `from` : `SlashCommandHint`

Screenshot 2024-03-27 at 12 45 09 AM

Fixes #32075
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!IMPORTANT]
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/8465758601>
> Commit: `cff14b0006441274e182d17ff66603219971ddbd`
> Cypress dashboard url: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=8465758601&attempt=2" target="_blank">Click here!</a>
> All cypress tests have passed 🎉🎉🎉

## Summary by CodeRabbit

- **New Features**
	- Enhanced the code editor with improved command suggestions and quick command generation, including URL and event parameter handling.
	- Introduced new styling for command suggestions in the code editor.
- **Refactor**
	- Updated functions and components for better handling and mapping of plugins within the code editor.
- **Chores**
	- Added necessary imports and updated the usage of various components and utilities to support new features and improvements.

### CODE REVIEW

**Review by hetunandu — DISMISSED** [review]

File: app/client/src/components/editorComponents/CodeEditor/generateQuickCommands.tsx [hunk 1]
```diff
@@ -97,17 +104,46 @@ export function Command(props: {
...
+      event.stopPropagation();
+      if (!props.url) return;
+      PerformanceTracker.startTracking(PerformanceTransactionName.OPEN_ACTION, {
+        url: props.url,
+      });
+      history.push(props.url, { invokedBy: NavigationMethod.SlashCommandHint });
+      AnalyticsUtil.logEvent("EDIT_ACTION_CLICK", props.eventParams || {});
+    },
+    [props.url],
+  );
```
[the remark this answers is not in the document]
Comment (sneha122) [reply app/client/src/components/editorComponents/CodeEditor/generateQuickCommands.tsx]: Thank you!! resolved the comment

**Review by hetunandu — DISMISSED** [review]

**Review by rishabhrathod01 — APPROVED** [review]
Code changes look good to me.

**Pull request merged**